packages/report-server/src/routes/fetchReport.ts

```diff
@@ -19,20 +19,30 @@ const getFilterFromReq = (req: ReportsRequest): FetchReportQuery => {
 
 class FetchReportRouteHandler {
   fetchReport = async (req: ReportsRequest, res: Response): Promise<void> => {
-    const { params, models, accessPolicy, body } = req;
+    const { params, query, models, accessPolicy, body } = req;
     const filter = getFilterFromReq(req);
     const aggregator = createAggregator(Aggregator, {
       session: { getAuthHeader: () => req.headers.authorization },
     });
     const reportBuilder = new ReportBuilder();
     if (body.testConfig) {
+      const { permissionGroup: permissionGroupName } = query;
+      if (permissionGroupName) {
+        await checkUserHasAccessToReport(
+          models,
+          accessPolicy,
+          permissionGroupName,
+          filter.organisationUnitCodes.split(','),
+        );
+      }
```

Comment:
This is to allow permission check when fetch preview data

Reply:
Hehe I'm glad someone finally added a permission check to here! The security hole was giving me anxiety 🙀 

We do however have to take a further look at the permissions model for this endpoint (and the preview functionality in the viz builder in general) to ensure that everything is locked down!

Even this isn't quite secure, since we aren't able to do any checks on the data elements the user is requesting, so they can query for whatever data they like, and as long as they have some permissions to that country, they can get the data (regardless if users with those permissions should have access to that data...)